common/flatpak-dir.c

```diff
@@ -4946,8 +4946,31 @@ flatpak_dir_create_system_child_repo (FlatpakDir   *self,
         }
     }
 
-  /* Ensure the config is updated */
   config = ostree_repo_copy_config (new_repo);
+
+  /* Verify that the mode is the expected one; if it isn't, recreate the repo */
+  current_mode = g_key_file_get_string (config, "core", "mode", NULL);
+  if (current_mode == NULL || g_strcmp0 (current_mode, parent_mode) != 0)
+    {
+      flatpak_rm_rf (repo_dir, NULL, NULL);
+
+      g_object_unref (repo_dir);
+      repo_dir = g_file_get_child (cache_dir, tmpdir_name);
```

Comment:
Why do you need to do this, repo_dir is just a GFile, i.e. a pathname. It should not be affected by removing it.